### PR TITLE
[MRKT-22, MRKT-23]: implements artist header with API data

### PR DESCRIPTION
**Pull request by scandycuz**

### CODE REVIEW

**Comment by scandycuz** [comment]
Marking as a draft, as the artist API endpoint needs to be updated to return the banner image url.

File: apps/marketplace/src/components/BannerImage.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    `(max-width: ${theme.breakpoints.values.sm}px)`
+  );
+
+  const resizedCoverImage = resizeCloudinaryImage(imageUrl, {
+    height: 200,
+    width: 1600,
+  });
+
+  if (isLoading) {
+    return <Box height={ 200 } width="100%" />;
```
Comment (scandycuz) [inline apps/marketplace/src/components/BannerImage.tsx:26]: Went with not rendering a loading skeleton here, as they are transparent, and the profile image skeleton on top of it looked a bit strange overlapping it.

File: apps/marketplace/src/app/artist/[artistId]/page.tsx [hunk 2]
```diff
@@ -13,48 +13,31 @@ interface ArtistProps {
...
-          sx={ {
-            backgroundImage: `url(${resizedCoverImage})`,
-            backgroundPosition: "center",
-            backgroundSize: "cover",
-            height: isBelowSmBreakpoint ? "130px" : "200px",
-            width: "100%",
-          } }
-        />
+        { /* TODO: replace with banner image url when available */ }
+        <BannerImage imageUrl={ "" } isLoading={ isLoading } />
```
Comment (scandycuz) [inline apps/marketplace/src/app/artist/[artistId]/page.tsx:31]: Planning on implementing this in its own PR when the API is updated.

File: packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        <Stack
+          alignItems={ ["center", "center", "flex-start"] }
+          direction="column"
+          px={ 5 }
+          spacing={ 1 }
+        >
+          <Skeleton height={ 36 } variant="rectangular" width={ 280 } />
+          <Skeleton height={ 12 } variant="rectangular" width={ 120 } />
+
+          { !isBelowMdBreakpoint && (
```
Comment (dmkirshon) [inline packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx:39]: Looks like `useBetterMediaQuery` hook seems to lag in activation when in a mobile view as seen below in the skeleton render. First render is false/null and then second render activates.

https://github.com/projectNEWM/newm-web/blob/653f3c5f740edac42c22d9fda8c3079b9073f9ec/packages/utils/src/lib/hooks.ts#L325

Solution Found via: https://github.com/chakra-ui/chakra-ui/issues/3580#issuecomment-1853816465

Fix the `useBetterMediaQuery` hook with :
```  
const [matches, setMatches] = useState<boolean | null>( window.matchMedia(mediaQueryString).matches );
  ```
  
  https://github.com/projectNEWM/newm-web/assets/86050631/514920b4-41cc-4e6e-b52a-2cc264d4f9bd

File: apps/studio/src/pages/home/owners/Owner.tsx [hunk 4]
```diff
@@ -74,11 +74,10 @@ const Owner: FunctionComponent = () => {
       />
 
       <ProfileHeader
-        firstName={ firstName }
         isVerified={ isVerified }
-        lastName={ lastName }
         location={ location }
-        profileImageUrl={ pictureUrl }
+        name={ `${firstName} ${lastName}` }
```
Comment (dmkirshon) [inline apps/studio/src/pages/home/owners/Owner.tsx:79]: Is there already a preference between Studio nickname and actual name?

File: apps/marketplace/src/components/skeletons/SalesSkeleton.tsx
Comment (dmkirshon) [inline apps/marketplace/src/components/skeletons/SalesSkeleton.tsx:1]: Not specific on this skeleton, but for `SongCardSkeleton` child, the responsive view text skeletons are a bit overlapping.

image

File: apps/marketplace/src/components/skeletons/SalesSkeleton.tsx [hunk 5]
```diff
@@ -15,7 +15,11 @@ const SalesSkeleton: FunctionComponent<SalesSkeletonProps> = ({
     <Stack alignItems="center">
       { hasTitle && (
         <Box mb={ 3.5 }>
-          <Skeleton height={ 76 } width={ 480 } />
+          <Skeleton
+            sx={ { fontSize: ["24px", "24px", "32px"] } }
+            variant="rectangular"
+            width={ 480 }
```
Comment (dmkirshon) [inline apps/marketplace/src/components/skeletons/SalesSkeleton.tsx:21]: The width can probably be responsive as well?

File: apps/marketplace/src/components/skeletons/ArtistsSkeleton.tsx [hunk 6]
```diff
@@ -17,7 +17,11 @@ const SalesSkeleton: FunctionComponent<SalesSkeletonProps> = ({
     <Stack alignItems="center">
       { hasTitle && (
         <Box mb={ 3.5 }>
-          <Skeleton height={ 76 } width={ 480 } />
+          <Skeleton
+            sx={ { fontSize: ["24px", "24px", "32px"] } }
+            variant="rectangular"
+            width={ 480 }
```
Comment (dmkirshon) [inline apps/marketplace/src/components/skeletons/ArtistsSkeleton.tsx:23]: Same as `SalesSkeleton`, width can probably be adjusted for responsiveness?

File: packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          ) }
+        </Stack>
+      </Stack>
+
+      <Box mt={ 2.5 }>
+        <SocialsSkeleton numItems={ 6 } />
+      </Box>
+
+      { isBelowMdBreakpoint && (
+        <Box alignSelf="stretch" pt={ 1.5 }>
```
Comment (dmkirshon) [inline packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx:52]: This wasn't stretching for me on the responsive view, the Skeleton width below should probably be removed or set to `100%`.

File: apps/marketplace/src/app/artist/[artistId]/page.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
 "use client";
-import { Box, Container, Stack, Typography, useTheme } from "@mui/material";
+import { Container, Stack, Typography, useTheme } from "@mui/material";
 import { FunctionComponent, useState } from "react";
 import { resizeCloudinaryImage, useBetterMediaQuery } from "@newm-web/utils";
```
Comment (dmkirshon) [inline apps/marketplace/src/app/artist/[artistId]/page.tsx]: `useTheme`, `resizeCloudinaryImage`, `useBetterMediaQuery` are all marked as unused

File: apps/marketplace/src/app/artist/[artistId]/page.tsx [hunk 9]
```diff
@@ -63,16 +46,16 @@ const Artist: FunctionComponent<ArtistProps> = ({ params }) => {
...
-          isOpen={ isAboutModalOpen }
-          name={ `${mockArtist.firstName} ${mockArtist.lastName}` }
-          socials={ socials }
-          onClose={ () => setIsAboutModalOpen(false) }
-        >
-          <Typography variant="subtitle1">{ mockArtist.description }</Typography>
-        </ProfileModal>
       </Stack>
+
+      <ProfileModal
```
Comment (dmkirshon) [inline apps/marketplace/src/app/artist/[artistId]/page.tsx:50]: The Modal is off Center, not sure if this on purpose or from certain flex settings, Didn't look at the full impact in Studio but it did center by adding `justifySelf: "center"` to `sx` in the `Modal.tsx` component.

https://github.com/projectNEWM/newm-web/blob/653f3c5f740edac42c22d9fda8c3079b9073f9ec/packages/elements/src/lib/Modal.tsx#L39

**Review by dmkirshon — APPROVED** [review]
Nothing blocking but a few comments/suggestions for refactor and to improve UI rendering. Otherwise LGTM!

File: apps/marketplace/src/app/artist/[artistId]/page.tsx (hunk 8 above)
Comment (scandycuz) [reply apps/marketplace/src/app/artist/[artistId]/page.tsx]: Strange the warnings no longer appear in the console where the app is run. Removed the imports.

File: apps/marketplace/src/app/artist/[artistId]/page.tsx (hunk 9 above)
Comment (scandycuz) [reply apps/marketplace/src/app/artist/[artistId]/page.tsx:50]: Good catch, added styling to have the modal content have a minimum width of the container. The shorter text content was causing it to not be the full width.

File: apps/marketplace/src/components/skeletons/SalesSkeleton.tsx
Comment (scandycuz) [reply apps/marketplace/src/components/skeletons/SalesSkeleton.tsx:1]: Good catch. Added styling so the skeletons don't overflow their containers.

File: apps/marketplace/src/components/skeletons/SalesSkeleton.tsx (hunk 5 above)
Comment (scandycuz) [reply apps/marketplace/src/components/skeletons/SalesSkeleton.tsx:21]: Added styling to not overflow the container here as well.

File: apps/studio/src/pages/home/owners/Owner.tsx (hunk 4 above)
Comment (scandycuz) [reply apps/studio/src/pages/home/owners/Owner.tsx:79]: The header component was already concatenating the first and last name and displaying it. This just accounts for the marketplace data only providing a `name` prop.

File: packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx (hunk 3 above)
Comment (scandycuz) [reply packages/components/src/lib/skeletons/ProfileHeaderSkeleton.tsx:39]: Nice, good find.

**Pull request merged**